### DISCUSSION TITLE
VSCode Extension

### DISCUSSION THREAD

**Discussion by konradr** [gitbutlerapp/gitbutler#2605] [Ideas]
Would love to see it as a vscode extension rather than stand alone app. Similar to GitLens

**Comment by vicary** [comment]
A standalone app is the only reason keeping me from using SourceTree and GitHub Desktop, there is enough cluttering on my desktop.

Would be a no brainer to try it out if it's a VS Code extension.

**Comment by leika** [comment]
I'dlike to second this. I realize some of the drag&drop UI Flow will be difficult, impossible or not fit well into vscode. But the branch management would fit well and it would be a great plugin IMHO. As a standaone option, I don't see enough value to leave my vscode or idea workflow just to maintain git ...

**Comment by wiggleforlife** [comment]
Same request, but with the Intellij family - would switch super fast

**Comment by schacon** [comment]
It's a good idea, but we would need some sort of API/server, which we don't quite have yet. But 💡

**Comment by orimay** [comment]
Would love this too! A standalone app is a bit too much

**Comment by ozyx** [comment]
A VSCode plugin to at least restore support for some of the features that break when using it. A "Share GitHub Permalink" context menu action is absolutely essential imo

**Comment by MartinBernstorff** [comment]
This is blocking my adoption as well; the context switch of using a separate app is too large for most of the work I do, but super excited about the prospect of GitButler!

**Comment by sidarth-23** [comment]
Adding my petition here. Now that project idx is open to public, there is no way for me to use gitbutler in there

**Comment by smoochy** [comment]
Any update on this?

**Reply by Byron** [reply]
There is no work going in this direction right now and it's not on the roadmap yet.
However, I [have created an issue](https://github.com/gitbutlerapp/gitbutler/issues/4128) to more easily interact with this request, and collection information as we go.

**Comment by g-kartik** [comment]
Second this

**Comment by alireza-taleghani-stratsys** [comment]
even some cli approach to open its UI and pointing project to the current repo would be a nice thing to have

**Comment by antspy** [comment]
Let me also add - I use VSCode on my machine, but VScode connects to a remote VM and all the development happens there. My local machine doesn't even have git installed. So I physically cannot try this product unless it's integrated with vscode, or alternatively I can run it as a binary on my remote VM and connect to it using the browser.

**Reply by Byron** [reply]
Thanks for sharing! In theory, thanks to `tauri` and the application architecture, having the backend run on a remote machine can technically already work, after all this is how the UI communications to the backend on the same host already.
However, currently `tauri` has no mechanism to actually support that.

**Reply by antspy** [reply]
Thanks for the reply! Something to think about then :) I will be following!

**Comment by mhalavanja** [comment]
Hi, any plans for this feature? Perhaps somewhat easier now that there is a CLI :)
